internal/guest/runtime/hcsv2/container.go

```diff
@@ -69,6 +69,20 @@ func (c *Container) ExecProcess(ctx context.Context, process *oci.Process, conSe
 	// Add in the core rlimit specified on the container in case there was one set. This makes it so that execed processes can also generate
 	// core dumps.
 	process.Rlimits = c.spec.Process.Rlimits
+
+	// If the client provided a user for the container to run as, we want to have the exec run as this user as well
+	// unless the exec's spec was explicitly set to a different user. If the Username field is filled in on the containers
+	// spec, at this point that means the work to find a uid:gid pairing for this username has already been done, so simply
+	// assign the uid:gid from the container.
+	if process.User.Username != "" {
+		// The exec provided a user string of it's own. Grab the uid:gid pairing for the string (if one exists).
+		if err := setUserStr(&oci.Spec{Root: c.spec.Root, Process: process}, process.User.Username); err != nil {
```

Comment:
Yep! Or add a new one that takes just the process spec and not the whole thing

Reply:
Would it be too much work to modify `setUserStr` (and its callers) to just take process spec and rootfs path? I am a bit worried about passing this `Spec` created on the fly here. Any functions that are called in the call chain of `setUserStr` (now & in the future) will use this spec and expect to see valid values in there. That can cause bugs that will be difficult to debug.

Reply:
Also, do we need to clear the `Username` field once we are done processing it?

Reply:
> Also, do we need to clear the `Username` field once we are done processing it?

It's ignored by runc so probably not. We could if we want to though. Would you prefer?

> Would it be too much work to modify `setUserStr` (and its callers) to just take process spec and rootfs path? I am a bit worried about passing this `Spec` created on the fly here. Any functions that are called in the call chain of `setUserStr` (now & in the future) will use this spec and expect to see valid values in there. That can cause bugs that will be difficult to debug.

No this wouldn't be much work, I can change

Reply:
Thanks. Let us change the `setUserStr` definition to only take process spec and rootfs path then and also clear the `Username` field just to be sure that it won't be used anywhere else.

Reply:
Ok after reworking `setUserStr`, I'm not really thrilled about how much this actually changes haha. You basically have to have all of the functions it shells out to return a uid, guid, error pair now and then set that on the spec outside of the function which is a departure from the other similar functions `setProcess` , `setCoreRLimit`, `addLinuxDeviceToSpec` and friends. 

The only field the functions rely on being present is `spec.Root.Path`, which is being set directly from the container it's being launched ins spec and is guaranteed to be filled in, otherwise the container wouldn't have been able to launch. If someone did rework `setUserStr` down the road and cause some kind of nil deref, we have the functionality to log panic logs anyways so I don't think the error would be too hard to track down

Reply:
I can still clear the username field though

Reply:
I see. Never mind then. Let us keep the current behavior and just clear out the `username` field.